Userland/DevTools/Profiler/FlameGraphView.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, Nicholas Hollett <niax@niax.co.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Profile.h"
+#include <AK/Function.h>
+#include <AK/Optional.h>
+#include <LibGUI/Model.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGfx/Color.h>
+
+namespace Profiler {
+
+class FlameGraphView : public GUI::Widget
+    , GUI::ModelClient {
+    C_OBJECT(FlameGraphView);
+
+public:
+    virtual ~FlameGraphView() override;
+
+    Function<void()> on_hover_change;
+
+    GUI::ModelIndex const* hovered_index() const;
```

Comment:
Please use an invalid index (`ModelIndex()`) to signal nothing hovered, to match how the rest of the system is written.